### PR TITLE
Oep-18 compliance

### PR DESCRIPTION
**Pull request by morenol**
Changes to make a plugin oep-18 compliant:

* Add constraints.txt 
* Add pip-tools.in
* Add test.in
* Update `upgrade` target

### CODE REVIEW

**Comment by morenol** [comment]
I removed the .txt requirements (except by pip-tools.txt) they should be generated running `make upgrade` when the repo is initialized.

**Review by felipemontoya — APPROVED** [review]

File: {{ cookiecutter.project_slug }}/requirements/constraints.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# pin when possible.  Writing an issue against the offending project and
+# linking to it here is good.
+
+# Django version 2.0.0 dropped support for python 2.7
+Django<2.0
+
+# Already in python3 standard library
+futures; python_version == "2.7"
+
+# TODO: Add contraint explanation
```
Comment (Squirrel18) [inline {{ cookiecutter.project_slug }}/requirements/constraints.txt:17]: Typo: constraint.

File: {{ cookiecutter.project_slug }}/Makefile [hunk 2]
```diff
@@ -17,11 +17,18 @@ clean: ## delete most git-ignored files
 	find . -name '*~' -exec rm -f {} +
 
 requirements: ## install environment requirements
-	pip install -r requirements.txt
+	pip install -r requirements/base.txt
 
+# Define PIP_COMPILE_OPTS=-v to get more information during make upgrade.
+PIP_COMPILE = pip-compile --rebuild --upgrade $(PIP_COMPILE_OPTS)
```
Comment (Squirrel18) [inline {{ cookiecutter.project_slug }}/Makefile:23]: Should we define it at the top of the file? @morenol

File: {{ cookiecutter.project_slug }}/requirements/pip-tools.in [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Just the dependencies to run pip-tools, mainly for the "upgrade" make target
```
Comment (Squirrel18) [inline {{ cookiecutter.project_slug }}/requirements/pip-tools.in:1]: Why do we need this file? Can we just install pip-tools in the upgrade make target? @morenol

Comment (morenol) [reply {{ cookiecutter.project_slug }}/requirements/pip-tools.in:1]: Yes, we can do that, but then we will have problems if a new version of pip-tools is not backward compatible

**Pull request merged**